test/extended/dr/quorum_restore.go

```diff
@@ -92,63 +83,37 @@ var _ = g.Describe("[sig-etcd][Feature:DisasterRecovery][Disruptive]", func() {
 				framework.Logf("Verify SSH is available before restart")
 				masters := masterNodes(oc)
 				o.Expect(len(masters)).To(o.BeNumerically(">=", 1))
-				survivingNode := masters[rand.Intn(len(masters))]
-				survivingNodeName := survivingNode.Name
-				expectSSH("true", survivingNode)
 
 				err = scaleEtcdQuorum(oc.AdminKubeClient(), 0)
 				o.Expect(err).NotTo(o.HaveOccurred())
 
 				expectedNumberOfMasters := len(masters)
-				survivingMachineName := getMachineNameByNodeName(oc, survivingNodeName)
-				survivingMachine, err := ms.Get(context.Background(), survivingMachineName, metav1.GetOptions{})
 				o.Expect(err).NotTo(o.HaveOccurred())
 
-				framework.Logf("Destroy %d masters", len(masters)-1)
-				var masterMachines []string
-				for _, node := range masters {
-					masterMachine := getMachineNameByNodeName(oc, node.Name)
-					masterMachines = append(masterMachines, masterMachine)
+				firstMaster := masters[0]
+				framework.Logf("Make etcd backup on first master %v", firstMaster)
+				expectSSH("sudo -i /bin/bash -cx 'rm -rf /home/core/backup; /usr/local/bin/cluster-backup.sh ~core/backup'", firstMaster)
 
-					if node.Name == survivingNodeName {
+				for i, node := range masters {
+					if i == 0 {
 						continue
 					}
+					framework.Logf("Stopping etcd and kube-apiserver pods and removing data-dir from %s", node.Name)
+					expectSSH("sudo -i /bin/bash -cx 'mv /etc/kubernetes/manifests/etcd-pod.yaml /tmp'", node)
 
-					framework.Logf("Destroying %s", masterMachine)
-					err = ms.Delete(context.Background(), masterMachine, metav1.DeleteOptions{})
-					o.Expect(err).NotTo(o.HaveOccurred())
+					time.Sleep(180 * time.Second)
+
+					expectSSH("sudo -i /bin/bash -cx 'mv /etc/kubernetes/manifests/kube-apiserver-pod.yaml /tmp; rm -rf /var/lib/etcd'", node)
```

Comment:
kube-apiserver takes ~70 seconds to shutdown.
Should there be some waiting after the manifest file is moved ?

Reply:
in our tests, we found reversing order created another failure case that required kubelet to restart to un wedge. I will add same wait afterward as that does make sense. I will test the ordering change locally.

Reply:
@tedyu first local run of this test with etcd and apiserver reversed order failed. Signature is a little different. New Pods are in `Pending` status indefinitely.

```NAME                                                           READY   STATUS      RESTARTS   AGE
etcd-ip-10-0-129-105.us-west-1.compute.internal                1/1     Running     0          26m
etcd-ip-10-0-135-219.us-west-1.compute.internal                4/4     Running     1          103m
installer-2-ip-10-0-129-105.us-west-1.compute.internal         0/1     Completed   0          102m
installer-2-ip-10-0-135-219.us-west-1.compute.internal         0/1     Completed   0          103m
installer-2-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          103m
installer-3-ip-10-0-203-76.us-west-1.compute.internal          0/1     Pending     0          25m
revision-pruner-2-ip-10-0-129-105.us-west-1.compute.internal   0/1     Completed   0          99m
revision-pruner-2-ip-10-0-135-219.us-west-1.compute.internal   0/1     Completed   0          99m
revision-pruner-2-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          99
```

```
installer-2-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          104m
installer-3-ip-10-0-129-105.us-west-1.compute.internal         0/1     Completed   0          103m
installer-3-ip-10-0-135-219.us-west-1.compute.internal         0/1     Completed   0          103m
installer-3-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          100m
installer-4-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          99m
installer-5-ip-10-0-129-105.us-west-1.compute.internal         0/1     Completed   0          96m
installer-5-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          98m
installer-6-ip-10-0-129-105.us-west-1.compute.internal         0/1     Completed   0          94m
installer-6-ip-10-0-135-219.us-west-1.compute.internal         0/1     Completed   0          92m
installer-7-ip-10-0-129-105.us-west-1.compute.internal         0/1     Completed   0          83m
installer-7-ip-10-0-135-219.us-west-1.compute.internal         0/1     Completed   0          91m
installer-7-ip-10-0-203-76.us-west-1.compute.internal          0/1     Completed   0          87m
installer-8-ip-10-0-203-76.us-west-1.compute.internal          0/1     Pending     0          26m
kube-apiserver-ip-10-0-129-105.us-west-1.compute.internal      4/4     Running     6          83m
kube-apiserver-ip-10-0-135-219.us-west-1.compute.internal      4/4     Running     0          91m
kube-apiserver-ip-10-0-203-76.us-west-1.compute.internal       4/4     Running     0          87m
revision-pruner-2-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          103m
revision-pruner-3-ip-10-0-129-105.us-west-1.compute.internal   0/1     Completed   0          103m
revision-pruner-3-ip-10-0-135-219.us-west-1.compute.internal   0/1     Completed   0          100m
revision-pruner-3-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          100m
revision-pruner-4-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          98m
revision-pruner-5-ip-10-0-129-105.us-west-1.compute.internal   0/1     Completed   0          94m
revision-pruner-5-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          97m
revision-pruner-6-ip-10-0-129-105.us-west-1.compute.internal   0/1     Completed   0          92m
revision-pruner-6-ip-10-0-135-219.us-west-1.compute.internal   0/1     Completed   0          91m
revision-pruner-7-ip-10-0-129-105.us-west-1.compute.internal   0/1     Completed   0          79m
revision-pruner-7-ip-10-0-135-219.us-west-1.compute.internal   0/1     Completed   0          87m
revision-pruner-7-ip-10-0-203-76.us-west-1.compute.internal    0/1     Completed   0          83m
```

Reply:
So what is interesting here is this pod is bogus.

> etcd-ip-10-0-135-219.us-west-1.compute.internal                4/4     Running     1          103m

If I delete just this single mirror pod everything completes as expected.

Reply:
moving comments to main thread